docs/troubleshooting/kubernetes.rst

```diff
@@ -1,13 +1,223 @@
+
+-----------------------------------------
+
+Why didn't the |kctlr| create any objects on the BIG-IP system?
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Here are a few things to check:
+
+Does the namespace of the Kubernetes resource match the namespace(s) you set the Controller to watch?
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+.. code-block:: yaml
+   :caption: Excerpt from a sample Deployment
+
+   apiVersion: extensions/v1beta1
+   kind: Deployment
+   metadata:
+     name: k8s-bigip-ctlr-deployment
+     namespace: kube-system
+   ...
+             args: [
+               "--bigip-username=$(BIGIP_USERNAME)",
+               "--bigip-password=$(BIGIP_PASSWORD)",
+               "--bigip-url=10.190.24.171",
+               "--bigip-partition=kubernetes",
+               "--namespace=<my-namespace>",
+               ]
+   ...
+
+
+Are the Service name and port provided in your virtual server ConfigMap correct?
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+.. code-block:: yaml
+   :caption: Sample Kubernetes Service
+   :emphasize-lines: 4, 11
+
+   kind: Service
+   apiVersion: v1
+   metadata:
+     name: hello
+   spec:
+     selector:
+       app: hello
+       tier: backend
+     ports:
+     - protocol: TCP
+       port: 80
+       targetPort: http
+
+*Source:* `Connect a Front End to a Back End Using a Service <https://kubernetes.io/docs/tasks/access-application-cluster/connecting-frontend-backend/#creating-the-backend-service-object>`_
+
+.. code-block:: yaml
+   :caption: Excerpt from a sample virtual server ConfigMap
+   :emphasize-lines: 10-11
+
+   kind: ConfigMap
+   apiVersion: v1
+   ...
+   data:
+    schema: "f5schemadb://bigip-virtual-server_v0.1.4.json"
+     data: |
+       {
+         "virtualServer": {
+           "backend": {
+             "servicePort": 80,
+             "serviceName": "hello",
+           },
+      ...
+       }
+
+Does the `service type`_ provided in the BIG-IP Controller Deployment match the service type defined for the Service?
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+- The default service type used by Kubernetes is ``clusterIP``.
+- The default service type used by the |kctlr| is ``nodeport``.
+
+**If you didn't specify a service type** in the Service definition or the |kctlr| Deployment, you probably have a service type mismatch. See :ref:`Nodeport mode vs Cluster mode <kctlr modes>` for more information about each service type and recommended use.
+
+
+Did you provide valid JSON?
+^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The |kctlr| can only parse valid JSON. Run your desired configurations through a JSON linter before use to avoid potential object creation errors.
+
+
+Have you used the correct version of the `F5 schema`_ to support your version of the BIG-IP Controller?
```

Comment:
For this section, I'd recommend removing the link of `F5 schema` from the bolded title and moved to the second line. 

This may not be the right place, but I think we should add an F5 schema to CC version table. How does a reader know which schema version maps to a CC.

Reply:
@rsalsa We do have a schema versioning table now: http://clouddocs.f5.com/containers/v2/releases_and_versioning.html#f5-schema

Reply:
Ah, not sure how I missed that. 

Can we link directly to this URL instead of sending them to the GH?